### DISCUSSION TITLE
Can't set compositeDeletePolicy

### DISCUSSION THREAD

**Discussion by fabioaraujopt** [crossplane/crossplane#3873] [General]
We are building composite resources (eg. "Application") by running the following code `kubectl crossplane build configuration` and `kubectl crossplane push configuration "ecr:xxxx"` inside a folder containing the following files:

- compositeresourcedefinition.yaml (containing a `kind:CompositeResourceDefinition` CRD)
- composition.yaml (containing a `kind:Composition` type CRD)
- crossplane.yaml (containing a `kind:Configuration` type CRD)

the CRDs get's installed and when we need to claim and create an application we just need to create a CRD inside the cluster of `kind: Application`

Now we found that when we delete the application some garbage remains in the cluster, we want to change the composition deletion to be Foreground (cascating).

So I updated crossplane to latest version `1.11.0` and then patched all Applications CRD by adding the field `spec.compositeDeletePolicy`. 

However I keep receiving the error ` strict decoding error: unknown field "spec.compositeDeletePolicy"`

I tried to update the providers, recreate the compositions, reinstall (remove and add the configuration) but nothing works, any ideia?

**Comment by bobh66** [comment]
compositeDeletePolicy is not a valid field on the XRD - it's only valid on Claims.

This feature is in development - https://github.com/crossplane/crossplane/issues/3744 - which will add a defaultCompositeDeletePolicy to the XRD so that if the Claim does not specify the compositeDeletePolicy it can take a default value from the XRD.  This is targeted for 1.12 which I think is due to be released in April.

**Reply by fabioaraujopt** [reply]
So If I add this to the compositeresourcedefinition as a spec field, it will work?

**Reply by bobh66** [reply]
If the feature makes it into Crossplane 1.12 then there will be a new `defaultCompositeDeletePolicy` attribute defined on all XRDs which can be used to set the `defaultCompositeDelete` policy to `Foreground`.  There is currently no way to do this at the XRD level.
